pallets/inflation-rewards/src/lib.rs

```diff
@@ -86,40 +88,52 @@ pub mod pallet {
             // Get the number of chains at this block (tanssi + container chain blocks)
             weight += T::DbWeight::get().reads_writes(1, 1);
             let registered_para_ids = T::ContainerChains::current_container_chains();
-            let number_of_chains: BalanceOf<T> =
-                ((registered_para_ids.len() as u32).saturating_add(1)).into();
-
-            // Issue new supply
-            let new_supply =
-                T::Currency::issue(T::InflationRate::get() * T::Currency::total_issuance());
-
-            // Split staking reward portion
-            let total_rewards = T::RewardsPortion::get() * new_supply.peek();
-            let (rewards_credit, reminder_credit) = new_supply.split(total_rewards);
-
-            let rewards_per_chain: BalanceOf<T> = rewards_credit.peek() / number_of_chains;
-            let (mut total_reminder, staking_rewards) = rewards_credit.split_merge(
-                total_rewards % number_of_chains,
-                (reminder_credit, CreditOf::<T>::zero()),
-            );
-
-            // Deposit the new supply dedicated to rewards in the pending rewards account
-            if let Err(undistributed_rewards) =
-                T::Currency::resolve(&T::PendingRewardsAccount::get(), staking_rewards)
-            {
-                total_reminder = total_reminder.merge(undistributed_rewards);
+
+            let mut number_of_chains: BalanceOf<T> = (registered_para_ids.len() as u32).into();
+
+            // We only add 1 extra chain to number_of_chains if we are
+            // in a parachain context with an orchestrator configured.
+            if let Some(_) = T::GetSelfChainBlockAuthor::get_block_author() {
```

Comment:
This is style improvement suggestion. Feel free to ignore it: 
If you can have a bool at the start like 
```
let is_solo_chain = T::GetSelfChainBlockAuthor::get_block_author().is_none();
```
or 

```
let is_orchestrator_chain = T::GetSelfChainBlockAuthor::get_block_author().is_some()
```
Whichever is suitable.

Then this would be much more clearer and you might not need the comment here as well as in the last calculation of `Weight`.

Reply:
Thanks for the suggestion! Although to be honest, I don't think it's worth applying it, as even if it would worth applying it in the first case, in the second case, I need to pass the `orchestrator_author` param to `reward_orchestrator_author` function anyways, and I don't want to use unwrap, so I don't see much sense on applying it only for the first case :)

Reply:
_disregard_